src/rpc/blockchain.cpp

```diff
@@ -2461,6 +2477,11 @@ static RPCHelpMan scanblocks()
         g_scanfilter_should_abort_scan = true;
         return true;
     } else if (request.params[0].get_str() == "start") {
+        {
+            LOCK(cs_relevant_blocks);
+            relevant_blocks = UniValue(UniValue::VARR);
```

Comment:
In commit "rpc: add relevant_blocks to scanblocks status" (4c9dc4bbe6b23ae7dba7badbdb6dd6c3d0f74e08)

Clearing `relevant_blocks` before checking `g_scanfilter_in_progress` seems not very nice, because it means if scanblocks start is called while another scan is in progress, instead of just returning an "already in progress" error, it will do that but also partially erase the state of the in-progress scan.

This behavior is not just a limitation of the new feature, but also a regression, because it messes up the "relevant_blocks" return value of the other RPC call when previously it would be accurate.

I think it should be straightforward to fix this by moving this new code below the reserve() call below.

Reply:
Good catch. Verified locally that truncation occurs. Fixed for next update.